jsonpb/jsonpb.go

```diff
@@ -116,13 +127,43 @@ type wkt interface {
 	XXX_WellKnownType() string
 }
 
+// customJSON can be implemented by messages that have custom non-object JSON
+// representations. If the message implements this method and the method
+// returns true, the JSON representation will not be an object (for example,
+// the message marshals itself as a string).
+type customJSON interface {
+	XXX_CustomJSON() bool
```

Comment:
I suspect there will be opinions on what this method should be named, and possibly even whether this should even be supported (custom message type that can marshal itself into custom JSON).

The main case I can think of is a dynamic message that happens to have a well-known type as a nested field. In that case, depending on implementation, it may be possible that the well-known type is actually represented via a dynamic message vs. the generated message type. Perhaps there are others. Possibly worth pointing out that the doc for `Any` just talks about messages with custom JSON and doesn't specifically restrict this to only well-known types.